### PR TITLE
fix(ComponentDidUpdate): Prevent animation to restart

### PR DESCRIPTION
**Pull request by junioramilson**
Animation restart only when it get finished

### CODE REVIEW

**Comment by cheng-kang** [comment]
@AmilsonJunior Thank you very much for the PR!

Could you please explain more about what issue this PR is trying to solve?

**Comment by junioramilson** [comment]
@cheng-kang When componentDidUpdate is called before the animation get finished `animation.start()` is called again causing an unexpected interruption in the animation, then it starts again before it get finished.

File: index.js [hunk 1]
```diff
@@ -59,25 +59,23 @@ export default class MarqueeLabel extends Component {
     } else {
       if (animation === null) {
         this.animatedTransformX.setValue(bgViewWidth);
-        this.setState({
+        await this.setState({
```
Comment (cheng-kang) [inline index.js]: According to [Component API - setState](https://shripadk.github.io/react/docs/component-api.html#setstate) and [source code](https://github.com/facebook/react/blob/master/packages/react/src/ReactBaseClasses.js#L53), `setState` does not return a Promise, so `await` is not doing anything here. 😉

But it does accept an optional callback.

**Review by cheng-kang — CHANGES_REQUESTED** [review]

**Comment by cheng-kang** [comment]
@AmilsonJunior Also could you provide minimum steps to reproduce the issue?

Once I could reproduce the issue, I will start to test the code, and hopefully get it merged soon!

**Comment by junioramilson** [comment]
@cheng-kang Try to test MarqueeLabel in a View that have constantly updates on render(). And why stop the animation on text change? Would be cool to update text and continue the animation. I'm using MarqueeLabel  on Market Data screen. :)

**Review by cheng-kang — APPROVED** [review]

**Comment by cheng-kang** [comment]
@AmilsonJunior thanks for your effort!

I will publish new version to NPM later today.

**Pull request merged**